### PR TITLE
Allows ordering bar drinks from Dusky

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[content]

## What this does
utilises code first made in #33966 and adds it to dusky, the robot serving the mining bar, limited to the drinks selection, price per each item 5-25.image
also adds functionality to make plasma based drinks for free if given plasma, like promised in one of the idle and welcome lines. plasma meat and plasma cabbages also work.
image
also adds messages for giving plasma ore, plasma coins, plasteel, plasma glass, plasma rifles, plasma tanks, plasma*men* (dead and alive) and plasmaFLOODING by mistake. plasma beakers and cutters can also be given to unlock plasma recipes but this won't dispense any drinks.

## Why it's good
gives it more purpose than just talking to.

## Changelog
:cl:
 * rscadd: You can now order drinks from robots serving bars in the mine. They'll make anything that can normally be made in the drinks dispensers behind them, alongside radium recipes, and plasma ones if you can find them some plasma... (sheet form only)

### CODE REVIEW

**Comment by Kromkar** [comment]
will this one horribly break something that shouldn't even be related? spin the wheel and find out!

**Comment by pedr69** [comment]
you ever hear about them goliaths? their hide is as strong as steel!

**Comment by Eneocho** [comment]
"Ah, a new face, eh? Last I saw one of you mining types must've been [ERROR: DATA OVERFLOW] days ago. Care for a drink? I make the meanest [ERROR: STRING MISSING] this side of the galaxy. Grab yourself a warm snack on that there microwave and sit down. Relax for a bit."

File: maps/randomvaults/mining/mining_surprise_items.dm [hunk 1]
```diff
@@ -59,4 +67,85 @@
...
+						"Aw, my favorite, I love brewin' with plasma. I think I'll make something here with it.",
+						"You found me some plasma? Well thank ya, here's a little something..."))
+				playsound(M.loc, pick('sound/items/polaroid1.ogg','sound/items/polaroid2.ogg'), 70, 1)
+				for(var/obj/O in sheets)
+					qdel(O)
+				for(var/obj/O2 in containers)
+					O2.reagents.del_reagent(PLASMA)
+				for(var/i in 1 to round(clamp(amount/10,1,5)))
+					// i could write something that finds every bar drink that has plasma in its recipe but let's face it,
+					// it's just this stuff and it's less expensive to look for these
```
Comment (Eneocho) [inline maps/randomvaults/mining/mining_surprise_items.dm:149]: while true, if anyone adds another plasma drink they'll also need to update this, and with no hint on the recipes page that this is the case they wouldn't

**Comment by ErrorsWindows** [comment]
can I get zpace cola
can I get zidecar
can I get cozmopolitan

**Comment by SECBATON-GRIFFON** [comment]
> can I get zpace cola can I get zidecar can I get cozmopolitan

this would've been a problem with the other vault too

**Pull request merged**